qemu/tests/hpt_miscellaneous.py

```diff
@@ -0,0 +1,97 @@
+import logging
+import re
+
+from virttest import error_context
+from virttest.utils_test.qemu import MemoryHotplugTest
+
+
+def set_hpt(session, test, hpt_size):
+    """Run this step to set HPT in guest"""
+    set_cmd = "echo %s > /sys/kernel/debug/powerpc/hpt_order"
+    st, o = session.cmd_status_output(set_cmd % (hpt_size))
+    # Get output after set command,some errors are there but
+    # they are acceptable
+    n_output = str(o)
+    if st:
+        # Invalid parameters,hpt_size < 18 and hpt_size > 46
+        if hpt_size < 18 or hpt_size > 46:
+            if not re.search(r"Invalid argument", n_output):
+                test.fail("Invalid param: '%s'" % n_output)
+        # Invalid parameter 0,it is handled in different way
+        # by kernel so dealing with it particularly
+        elif hpt_size == 0:
+            if not re.search(r"Input/output error", n_output):
+                test.fail("Invalid param zero : '%s'" % n_output)
+
+
+def verify_hpt(test, params, session, hpt_size):
+    """Run this step to check HPT in guest"""
+    get_cmd = "cat /sys/kernel/debug/powerpc/hpt_order"
+    if params.get("sub_type") != "negative":
+        s, get_hpt_value = session.cmd_status_output(get_cmd)
+        if s:
+            test.fail("Fail to get HPT value")
+        else:
+            if int(get_hpt_value) != hpt_size:
+                test.fail("HPT test failed")
```

Comment:
Suggest `test.fail("HPT order not match! '%s' vs '%s'" % (get_hpt_value, hpt_size))`